test/e2e/workspace_status_test.go

```diff
@@ -0,0 +1,290 @@
+
+		By("installing WorkspaceTemplate samples")
+		_, _ = fmt.Fprintf(GinkgoWriter, "Creating jupyter-k8s-shared namespace...\n")
+		cmd = exec.Command("kubectl", "create", "namespace", "jupyter-k8s-shared")
+		_, _ = utils.Run(cmd) // Ignore error if namespace already exists
+
+		By("applying production template for tests")
+		_, _ = fmt.Fprintf(GinkgoWriter, "Applying production template...\n")
+		cmd = exec.Command("kubectl", "apply", "-f",
+			"config/samples/workspace_v1alpha1_workspacetemplate_production.yaml")
+		_, err = utils.Run(cmd)
+		Expect(err).NotTo(HaveOccurred(), "Failed to create production template")
+	})
+
+	AfterAll(func() {
+		By("cleaning up test workspaces")
+		_, _ = fmt.Fprintf(GinkgoWriter, "Deleting test workspaces...\n")
+		cmd := exec.Command("kubectl", "delete", "workspace",
+			runningWorkspace, stoppedWorkspace, transitionWorkspace,
+			"--ignore-not-found", "--timeout=60s")
+		_, _ = utils.Run(cmd)
+
+		By("cleaning up test templates")
+		_, _ = fmt.Fprintf(GinkgoWriter, "Deleting test templates...\n")
+		cmd = exec.Command("kubectl", "delete", "workspacetemplate",
+			"production-notebook-template",
+			"--ignore-not-found", "--wait=false")
+		_, _ = utils.Run(cmd)
+
+		By("uninstalling CRDs")
+		_, _ = fmt.Fprintf(GinkgoWriter, "Uninstalling CRDs...\n")
+		cmd = exec.Command("make", "uninstall")
+		_, _ = utils.Run(cmd)
+
+		By("undeploying the controller-manager")
+		_, _ = fmt.Fprintf(GinkgoWriter, "Undeploying controller manager...\n")
+		cmd = exec.Command("make", "undeploy")
+		_, _ = utils.Run(cmd)
+
+		By("waiting for namespace to be fully terminated")
+		Eventually(func() error {
+			cmd := exec.Command("kubectl", "get", "namespace", "jupyter-k8s-system")
+			_, err := utils.Run(cmd)
+			if err != nil {
+				return nil // namespace doesn't exist, cleanup complete
+			}
+			return fmt.Errorf("namespace still exists")
+		}, 120*time.Second, 2*time.Second).Should(Succeed())
+		_, _ = utils.Run(cmd)
+	})
+
+	Context("Running State", func() {
+		It("should create resources when desiredStatus is Running", func() {
+			By("creating workspace with desiredStatus: Running")
+			cmd := exec.Command("kubectl", "apply", "-f",
+				"test/e2e/static/status-transitions/workspace-running.yaml")
+			_, err := utils.Run(cmd)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("verifying desiredStatus is Running")
+			verifyDesiredStatus(runningWorkspace, "Running")
+
+			By("verifying Valid condition becomes True")
+			verifyStatusCondition(runningWorkspace, controller.ConditionTypeValid, "True")
+
+			By("verifying Deployment is created")
+			verifyResourceExists("deployment", controller.GenerateDeploymentName(runningWorkspace), true)
+
+			By("verifying Service is created")
+			verifyResourceExists("service", controller.GenerateServiceName(runningWorkspace), true)
+
+			By("verifying Available condition is False initially")
+			verifyStatusCondition(runningWorkspace, controller.ConditionTypeAvailable, "False")
```

Comment:
should we wait till it be becomes available? or is that covered in another test?